lib/util/reddit.ts

```diff
@@ -0,0 +1,68 @@
+import * as centra from "centra";
+import { randomInt } from "crypto";
+import { RedditPost } from "../interfaces/reddit";
+
+/* API */
+
+/**
+ * Returns a random post/meme from a subreddit.
+ * @param subreddit Optional The subreddit string, defaults to a random english meme subreddit.
+ * @returns {RedditPost}
+ */
+ export async function getRandomPost(subreddit?: string): Promise<RedditPost | undefined> {
+	const url = `https://www.reddit.com/r/${subreddit ? subreddit : randomSubreddit()}/hot/.json?count=100`
+
+	return await buildPost(url)
+}
+
+
+/* Utils */
+
+// more languages coming soontm
+const randomSubreddits = {
+  en: ['memes', 'dankmemes', 'meirl'],
+}
+
+/**
+ * Gets a random subreddit from the list of available meme subreddits.
+ * @returns {string} The random subreddit string.
+ */
+function randomSubreddit(): string {
+	return randomSubreddits.en[randomInt(randomSubreddits.en.length)]
+}
+
+
+export function checkURL(url: string): boolean {
+  return url.match(/\.(jpeg|jpg|gif|png)$/) != null
+}
+
+export async function buildPost(url: string): Promise<RedditPost | undefined> {
+	try {
+		const response = await (
```

Comment:
use `Promise#then()`, you'll only need to await once

Reply:
This is how centra is implemented in the rest of the source code and I don't really think is necessary. Does this make any difference?

Reply:
stylistically yes, in practice not much. might be better to open a seperate PR

Reply:
oki